storage/blob.go

```diff
@@ -602,6 +617,37 @@ func (b BlobStorageClient) GetBlobProperties(container, name string) (*BlobPrope
 	}, nil
 }
 
+// SetBlobProperties replaces the ContentSetting for the specified blob.
+//
+// Some keys may be converted to Camel-Case before sending. All keys
+// are returned in lower case by GetBlobProperties. HTTP header names
+// are case-insensitive so case munging should not matter to other
+// applications either.
+//
+// See https://msdn.microsoft.com/en-us/library/azure/ee691966.aspx
+func (b BlobStorageClient) SetBlobProperties(container, name string, contentSetting ContentSetting) error {
+	params := url.Values{"comp": {"properties"}}
+	uri := b.client.getEndpoint(blobServiceName, pathForBlob(container, name), params)
+	headers := b.client.getStandardHeaders()
+
+	value := reflect.ValueOf(contentSetting)
+	for i := 0; i < value.NumField(); i++ {
```

Comment:
I think this logic is very valuable to be buried in `SetBlobProperties`, why don't you create a method called `headersFromStruct` that uses a field tag called "header"; (not "xml") and test it so that we can use it here and so on?